### PR TITLE
Add substitute for ES::Messaging::Reader

### PR DESCRIPTION
**Pull request by ntl**
I am using this to test projections in isolation.

### CODE REVIEW

File: lib/event_store/messaging/message/message__ref.rb [hunk 1]
```diff
@@ -1 +0,0 @@
-/Users/sbellware/projects/obsidian/event-store-messaging/lib/event_store/messaging/message.rb
```
Comment (sbellware) [inline lib/event_store/messaging/message/message__ref.rb]: What? You mean you don't get benefit from this hard link to a location on my file system?? ;)

File: lib/event_store/messaging/reader.rb [hunk 2]
```diff
@@ -4,6 +4,34 @@ class Reader < MessageReader
...
+
+        dependency :dispatcher, EventStore::Messaging::Dispatcher
+
+        def initialize
+          @messages = []
+        end
+
+        def self.build
+          new
+        end
```
Comment (ntl) [inline lib/event_store/messaging/reader.rb]: Notice I don't configure the dispatcher here. This is something I'm confused on. `build` is called by `subst-attr` with exactly 0 arguments, but I need the substitute to invoke methods on a `Projection`, which is using this substitute in place of its reader.

File: lib/event_store/messaging/reader.rb [hunk 3]
```diff
@@ -4,6 +4,34 @@ class Reader < MessageReader
       def self.http_reader
         Client::HTTP::Reader
       end
+
+      class Substitute
+        attr_reader :messages
+        attr_reader :reader
```
Comment (sbellware) [inline lib/event_store/messaging/reader.rb]: Hey, is this reader needed in the substitute?

Comment (ntl) [inline lib/event_store/messaging/reader.rb]: Removed, thanks!

**Comment by sbellware** [comment]
:green_apple: (looks good to merge)

**Pull request merged**